code/modules/projectiles/guns/projectile.dm

```diff
@@ -290,3 +324,33 @@
 		else
 			. += "No magazine inserted.<br>"
 	. += ..(FALSE)
+
+///Installs a new suppressor, assumes that the suppressor is already in the contents of src
+/obj/item/gun/projectile/proc/install_suppressor(obj/item/silencer/S)
+	suppressed = TRUE
+	w_class += S.w_class //Add our weight class to the item's weight class
+	suppressor = S
+	update_icon()
+
+/obj/item/gun/projectile/clear_suppressor()
+	if(!can_unsuppress)
+		return
+	if(istype(suppressor))
+		w_class -= suppressor.w_class
```

Comment:
```suggestion
		w_class = initial(w_class)
```

Reply:
This could potentially pose a problem if we add other attachments. Ideally the bayonet also uses this system.